scarb/src/compiler/compilers/test.rs

```diff
@@ -45,8 +45,11 @@ impl Compiler for TestCompiler {
                 add_statements_functions: unit
                     .compiler_config
                     .unstable_add_statements_functions_debug_info,
+                add_statements_code_locations: unit
+                    .compiler_config
+                    .unstable_add_statements_code_locations_debug_info,
             };
-            compile_test_prepared_db(db, config, main_crate_ids, test_crate_ids)?
+            compile_test_prepared_db(db, config, main_crate_ids, test_crate_ids, false)?
```

Comment:
Should be `unit.compiler_config.allow_warnings`. 

Btw I think the API starts to get messy here, cc @maciektr for when he comes back (I know Maciek were in progress of doing some changes to the API to make sure the devx is better)

Reply:
Thanks for raising it! I just dumped the draft here and make it compile, as I think @THenry14 already provided a PR for that, I will merge this into his pr.

Reply:
https://github.com/software-mansion/scarb/pull/1505